File: ant-sdk/antd-dart/example/06_private_data.dart
import 'dart:convert';
import 'dart:typed_data';

import 'package:antd/antd.dart';

/// Demonstrates storing and retrieving private (encrypted) data.
void main() async {
  final client = AntdClient();

  try {
    // Store private data — daemon returns the DataMap; caller keeps it.
    final data = Uint8List.fromList(utf8.encode('my secret data'));
    final result = await client.dataPut(data);
    print('Private data stored');
    print('  chunks: ${result.chunksStored}, mode: ${result.paymentModeUsed}');
    print('  data map: ${result.dataMap}');

    // Retrieve private data using the data map
    final retrieved = await client.dataGet(result.dataMap);
    print('Retrieved: ${utf8.decode(retrieved)}');
  } on AntdError catch (e) {
    print('Error: $e');
  } finally {
    client.close();
  }
}

File: ant-sdk/antd-rust/examples/06-private-data.rs
use antd_client::{Client, PaymentMode, DEFAULT_BASE_URL};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = Client::new(DEFAULT_BASE_URL);

    // Store private (encrypted) data
    let result = client
        .data_put(b"secret payload", PaymentMode::Auto)
        .await?;
    println!(
        "Private data stored (chunks: {}, mode: {})",
        result.chunks_stored, result.payment_mode_used
    );
    println!("Data map: {}", result.data_map);

    // Retrieve private data using the data map
    let data = client.data_get(&result.data_map).await?;
    println!("Decrypted: {}", String::from_utf8_lossy(&data));

    Ok(())
}
